flume-ng-core/src/test/java/org/apache/flume/client/avro/TestReliableSpoolingFileEventReader.java

```diff
@@ -112,6 +114,40 @@ private void deleteDir(File dir) {
   }
 
   @Test
+  public void testIncludePattern() throws IOException {
+    ReliableEventReader reader = new ReliableSpoolingFileEventReader.Builder()
+        .spoolDirectory(WORK_DIR)
+        .includePattern("^file2$")
+        .deletePolicy(DeletePolicy.IMMEDIATE.toString())
+        .build();
+ 
+    List<File> before = listFiles(WORK_DIR);
+    Assert.assertEquals("Expected 5, not: " + before, 5, before.size());
+
+    List<Event> events;
```

Comment:
There are a couple of copy-paste lines here and there, could you please refactor it a little bit?
e.g. extract the `reader.readEvents-reader.commit` loops to a method.
The assertion part (`listFiles`, checking the count, removing from the `expectedLeftFiles`, checking the emptiness of it) might be possible to extract too, to improve the readability of the code. (e.g. `void assertExpectedLeftFiles(String... expectedFileNames)`)